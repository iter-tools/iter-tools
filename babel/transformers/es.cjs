const { resolve } = require('path');
const babelJest = require('babel-jest');

module.exports = babelJest.createTransformer({
  configFile: resolve(__dirname, '../jest.config.cjs'),
});
