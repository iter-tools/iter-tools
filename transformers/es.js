const createTransformer = require('./create-transformer');

module.exports = createTransformer('babel-es.config.js');
