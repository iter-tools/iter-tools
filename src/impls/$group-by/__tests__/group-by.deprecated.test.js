/**
 * @generated-from ./$group-by.deprecated.test.js
 * This file is autogenerated from a template. Please do not edit it directly.
 * To rebuild it from its template use the command
 * > npm run generate
 * More information can be found in CONTRIBUTING.md
 */

import { groupBy } from '@iter-tools/es';
import { unwrapDeep } from '../../../test/helpers.js';

describe('groupBy (deprecated)', () => {
  it('returns source values grouped by identity', () => {
    const source = 'ABBAAC';
    const result = [
      ['A', ['A']],
      ['B', ['B', 'B']],
      ['A', ['A', 'A']],
      ['C', ['C']],
    ];
    expect(unwrapDeep(groupBy(null, source))).toEqual(result);
    expect(unwrapDeep(groupBy(null, source))).toEqual(result);
    expect(console.warn).callsMatchSnapshot();
  });
});
