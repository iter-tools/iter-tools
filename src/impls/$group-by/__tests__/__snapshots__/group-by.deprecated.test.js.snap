// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`groupBy (deprecated) returns source values grouped by identity 1`] = `
Array [
  Array [
    "\`groupBy(null, iterable)\` is deprecated and will be removed in iter-tools@8. Instead use group(iterable)",
  ],
]
`;
